client/src/pages/Details.tsx

```diff
@@ -1,3 +1,113 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { FaRegHeart } from 'react-icons/fa6';
+// import { IoMdHeart } from "react-icons/io";
+import { FaCircle } from 'react-icons/fa';
+import { FaRegCircle } from 'react-icons/fa';
+import { toDollars } from '../../lib/to-dollars';
+
+export type ProductDetails = {
+  productId: number;
+  imageUrl: string;
+  name: string;
+  price: number;
+  color: string;
+  sizes: string[];
+};
+
 export function Details() {
-  return <></>;
+  const { productId } = useParams<{ productId: string }>();
+  const [details, setDetails] = useState<ProductDetails | null>(null);
```

Comment:
I personally find it more natural to use `undefined` here instead of `null`. The code then looks like:
`const [details, setDetails] = useState<ProductDetails>();`
This works since the type definition of `useState` automatically adds `| undefined` to the type if you do not pass any parameters.

Below, you only check `!details` anyway, which works with either `null` or `undefined`.